state/postgres_types.go

```diff
@@ -1,4 +1,21 @@
 package state
 
+import "github.com/guregu/null"
+
 type Oid uint64
 type Xid uint32
+
+// PostgresType - User-defined custom data types
+type PostgresType struct {
+	Oid             Oid
+	DatabaseOid     Oid
+	SchemaName      string
+	Name            string
+	Type            string
+	UnderlyingType  null.String
+	NotNull         bool
+	Default         null.String
+	Constraint      null.String
```

Comment:
I wonder if we should prefix these four attributes with `Domain`?

In my understanding they are not used with the other types, and that makes it easier to understand what needs to be looked at for a given type.